client/src/components/Production/ProductionModal.jsx

```diff
@@ -0,0 +1,325 @@
+import {
+  Center,
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  FormControl,
+  FormLabel,
+  Input,
+  ModalFooter,
+  Select,
+  Textarea,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+} from '@chakra-ui/react';
+import { TextField } from '@material-ui/core';
+import { Fragment } from 'react';
+import useProductionModal from 'hooks/useProductionModal';
+
+const ProductionModal = () => {
+  const {
+    isSuccessPart,
+    dataPart,
+    formStyle,
+    elementStyle,
+    partType,
+    isOpen,
+    onOpen,
+    onClose,
+    form,
+    setForm,
+    productionRef,
+    colorRef,
+    quantityRef,
+    statusRef,
+    nameRef,
+    descriptionRef,
+    startDate,
+    endDate,
+    assemblyMachineRef,
+    noteRef,
+    qualityRef,
+    profitMarginRef,
+    gradeRef,
+    finishRef,
+    weightAmountRef,
+    weightTypeRef,
+    handleBarRef,
+    wheelRef,
+    chainRef,
+    frameRef,
+    pedalRef,
+    brakesRef,
+    seatRef,
+    forkRef,
+    handleStartDateInput,
+    handleEndDateInput,
+    handleSubmit,
+  } = useProductionModal();
+
+  const formPart = (
+    <>
+      <FormControl isRequired style={elementStyle}>
+        <FormLabel>Product</FormLabel>
+        <Select placeholder="Select production" ref={productionRef}>
+          {partType.map((part) => (
+            <Fragment key={part}>
+              <option value={part}>{part}</option>
+            </Fragment>
+          ))}
+        </Select>
+      </FormControl>
+      <div style={formStyle}>
+        <FormControl isRequired style={elementStyle}>
+          <FormLabel>Quality</FormLabel>
+          <Input ref={qualityRef} />
+        </FormControl>
+        <FormControl style={elementStyle}>
+          <FormLabel>Grade</FormLabel>
+          <Input ref={gradeRef} />
+        </FormControl>
+        <FormControl style={elementStyle}>
+          <FormLabel>Finish</FormLabel>
+          <Input ref={finishRef} />
+        </FormControl>
+      </div>
+    </>
+  );
+
+  const formBike = (
```

Comment:
seperate dat shit into more files

Reply:
@Mackbellemore i just used the `forwardRef`. let me know what you think of it.